### DISCUSSION TITLE
Blocked Common Types - Causes custom properties not to be migrated

### DISCUSSION THREAD

**Discussion by ProNotion** [Jumoo/uSyncMigrations#161] [Q&A]
I note that in the `SyncMigrationService` class there is a method called `PrepareContext` in which the following code exists:

```
if (options.BlockCommonTypes)
{
	context.AddBlocked(nameof(MediaType), UmbConstants.Conventions.MediaTypes.File);
	context.AddBlocked(nameof(MediaType), UmbConstants.Conventions.MediaTypes.Folder);
	context.AddBlocked(nameof(MediaType), UmbConstants.Conventions.MediaTypes.Image);
}
```
We have a couple of custom properties on the Image media type and as a result of this blocking those properties are not included in the migration. 

What would be the best way to introduce these properties as part of the migration process?

**Comment by AllanMT** [comment]
I assume this relates to Issue #92, where we had the same problem and I'd asked:
"Hi, I used your very-useful uSync.Migrations/4.0.0-phase.3 to try migrating an Umbraco 7.15.10 site uSync export to 10.5.1.
One issue we had was that an alttext property we had created on images, and two properties title and description that we had created for files (so that a parent page could display the titles and descriptions of the files in a document collection on the page) were in the v7 uSync export but not migrated in to 10. I wasn't sure if this was the sort of thing you'd want to add in to uSyncMigrations , or whether we should be adding something to do this (and if so, where we'd find out how)?"
At the moment, I manually create the new property in the Umbraco 10 back office, then look in the database to get the old and new propertytypeid, then run a statement like this, e.g. for our file titles, propertytypeid 89 becoming propertytypeid 109
INSERT INTO [areyouok-test10-14nov].[dbo].[umbracoPropertyData] (versionId, propertyTypeId, varcharValue)
SELECT v10.id, 109 AS propertyTypeId, d.dataNvarchar
FROM [areyouok-test7].[dbo].[cmsPropertyData] d, [areyouok-test7].[dbo].[umbracoNode] n, [areyouok-test7].[dbo].[cmsContentVersion] v, [areyouok-test10-14nov].[dbo].[umbracoNode] n10, [areyouok-test10-14nov].[dbo].[umbracoContentVersion] v10    
WHERE propertytypeid = 89 AND dataNvarchar IS NOT NULL AND dataNvarchar != '' AND d.contentNodeId = n.id AND d.versionId = v.VersionId AND n.uniqueID = n10.uniqueId AND n10.id = v10.nodeId AND v10.[current] = 1 
(and doing the same for the File Description property and Image Alt Text property.)

**Comment by KevinJump** [comment]
Hi, 

sorry super busy, so not really keeping up with this at the moment

these elements are blocked by default because Umbraco writes them should the media be their, so having them in the file and umbraco writing them causes false posistives and makes syncing a bit slower (when the media exists). - because things will flag as changes when they are not. 

however this is behind an option, (BlockCommonTypes) . so you shouldn't need to have your own version of the whole code base, you need a custom migration plan/profile (see examples https://github.com/Jumoo/uSyncMigrations/blob/main/MyMigrations/MyMigrationProfile.cs) and then you could set the BlockCommonTypes option to false, and it wouldn't add these blocks to the migration. 

Kevin

**Comment by ProNotion** [comment]
@KevinJump Thanks for the heads up, I can see that now. However, how would this work if your migration consists of multiple steps? For instance mine currently uses `UpgradeUmbracoSevenPlan`, followed by my own plan (so that I can set my preferred NestedContent Migrator, followed by the `BlockMigrationPlan`. Would I need to create my own plan for each and inherit from the core ones and then set the `BlockCommonTypes` option to false on each of them?
